test/test_dir.rs

```diff
@@ -0,0 +1,49 @@
+extern crate nix;
+extern crate tempdir;
+
+use nix::dir::{Dir, Type};
+use std::fs::File;
+use self::tempdir::TempDir;
+
+fn open(path: &::std::path::Path) -> nix::Result<Dir> {
```

Comment:
Would it be worthwhile to make this a method of `Dir`, so that you can open by either `RawFd` or a path?

Reply:
Done - both `fcntl::open` and `fcntl::openat` variants. Unlike my quick test version, I'm using a `NixPath`, and I'm expecting the caller to provide `OFlags`, so they can do `O_NOFOLLOW` or `O_EXLOCK` or whatever as desired.